core/src/main/scala/org/apache/spark/TaskEndReason.scala

```diff
@@ -212,9 +212,15 @@ case object TaskResultLost extends TaskFailedReason {
  * Task was killed intentionally and needs to be rescheduled.
  */
 @DeveloperApi
-case class TaskKilled(reason: String) extends TaskFailedReason {
+case class TaskKilled(
+    reason: String,
+    accumUpdates: Seq[AccumulableInfo] = Seq.empty,
+    private[spark] val accums: Seq[AccumulatorV2[_, _]] = Nil)
```

Comment:
Previously we use `AccumulableInfo` to expose accumulator information to end users. Now `AccumulatorV2` is already a public classs and we don't need to do it anymore, I think we can just do
```
case class TaskKilled(reason: String, accums: Seq[AccumulatorV2[_, _]])
```

Reply:
Yeah, I noticed `accumUpdates: Seq[AccumulableInfo]` is only used in JsonProtocol. Is that for a reason?

The current impl  is constructed to be sync with existing TaskEndReason such as `ExceptionFailure`
```
@DeveloperApi
case class ExceptionFailure(
    className: String,
    description: String,
    stackTrace: Array[StackTraceElement],
    fullStackTrace: String,
    private val exceptionWrapper: Option[ThrowableSerializationWrapper],
    accumUpdates: Seq[AccumulableInfo] = Seq.empty,
    private[spark] var accums: Seq[AccumulatorV2[_, _]] = Nil)
```

I'd prefer to keep in sync, leave two options for cleanup:
1. leave it as it is, then cleanup with ExceptionFailure together
2. Cleanup ExceptionFailure first.

@cloud-fan what do you think?

Reply:
let's clean up ExceptionFailure at the same time, and use only `AccumulatorV2` in this PR.

Reply:
@cloud-fan After a second look, I don't think we can clean up `ExceptionFailure` unless we can break `JsonProtocol`

Reply:
now the question is: shall we keep the unnecessary Seq[AccumulableInfo] in new APIs, to make the API consistent? I'd like to not keep the Seq[AccumulableInfo], we may deprecate it in the existing APIs in the near future.

Reply:
I'm ok with not keeping `Seq[AccumulableInfo]`. But it means inconsistent logic and api and may make future refactoring a bit difficult. 

Let's see what I can do.

>  I'd like to not keep the Seq[AccumulableInfo], we may deprecate it in the existing APIs in the near future.

BTW, I think we have already deprecated `AccumulableInfo`. Unless we are planing to remove it in Spark 3.0 and Spark 3.0 is the next release, `AccumulableInfo` will be there for a long time

Reply:
Hi @cloud-fan, I have looked at how to remove `Seq[AccumulableInfo]` tonight.
It turns out that we cannot because `JsonProtocol` calls `taskEndReasonFromJson` to reconstruct `TaskEndReason`s. Since `AccumulatorV2`  is an abstract class, we cannot simply construct `AccumulatorV2`s from json.

Even we are promoting `AccumulatorV2`,  we still need `AccumulableInfo` when (de)serializing json.

Reply:
I see, that makes sense, let's keep `AccumulableInfo`.

Reply:
@cloud-fan so, could you trigger the test and have a look?

And looks like I am not in the whitelist again...